pkg/kmm/kernelmapping.go

```diff
@@ -73,6 +57,38 @@ func (builder *KernelMappingBuilder) BuildKernelMappingConfig() (*moduleV1Beta1.
 	return builder.definition, nil
 }
 
+func (builder *KernelMappingBuilder) WithRegexp(regexp string) *KernelMappingBuilder {
+	if valid, _ := builder.validate(); !valid {
+		return builder
+	}
+
+	glog.V(100).Infof("Adding regexp %s to kernel mapping", regexp)
+
+	builder.definition.Regexp = regexp
+
+	if regexp == "" {
+		glog.V(100).Infof("The regex of NewRegExKernelMappingBuilder is empty")
+
+		builder.errorMsg = "'regexp' parameter can not be empty"
+	}
+
+	return builder
+}
+
+func (builder *KernelMappingBuilder) WithLiteral(literal string) *KernelMappingBuilder {
+	glog.V(100).Infof("Adding literal %s to the kernel mapping", literal)
+
+	builder.definition.Literal = literal
+
+	if literal == "" {
```

Comment:
Please move this validation before: 
builder.definition.Literal = literal
Update Literal only if builder.errorMsg == ""

Reply:
Same as https://github.com/openshift-kni/eco-goinfra/pull/105/files#r1316737580 with https://github.com/openshift-kni/eco-goinfra/blob/3c31a5ef4dea4b9dd46d392f9a15dff39672e058/pkg/kmm/kernelmapping.go#L49-L59

Reply:
The example you showed above is for NewBuilder* functions where we store error in order to return object anyway however for mutation functions With* we don't update object if argument is invalid.